psychopy/visual/vlcmoviestim.py

```diff
@@ -0,0 +1,625 @@
+    for your OS.
+
+    http://www.videolan.org/vlc/index.html
+
+2. pip install python-vlc
+
+To play a video:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create a new instance, `mov = visual.VlcMovieStim(..)`
+
+    shouldflip = mov.play()
+    continueRoutine = True
+    while continueRoutine:
+        if shouldflip:
+            # Draw other stimuli here
+            win.flip()
+        else:
+            time.sleep(0.001)
+        shouldflip = mov.draw()
+
+
+To fix "stale cache" VLC errors on Windows...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Libvlc has a bug related to changing from daylight standard time to daylight 
+saving time. After the time change libvlc complains that the plugin cache is 
+stale. The error message begins with "main libvlc error: stale plugins cache: 
+...". VLC provides an executable to update the plugin cache timestamps. To run 
+it, execute this as Administrator:
+
+    cd "C:\Program Files\VideoLAN\VLC"
+    vlc-cache-gen.exe "C:\Program Files\VideoLAN\VLC\plugins"
+
+
+Not implemented:
+~~~~~~~~~~~~~~~~
+
+* Horizontal/vertical flipping
+* getCurrentFrameTime
+* getPercentageComplete
+
+"""
+
+# Part of the PsychoPy library
+# Copyright (C) 2018 Jonathan Peirce
+# Distributed under the terms of the GNU General Public License (GPL).
+#
+# VlcMovieStim contributed by Dan Fitch, April 2019.
+# The MovieStim2 class was taken and rewritten to use only vlc
+
+from __future__ import absolute_import, division, print_function
+
+# If True then, on each flip a new movie frame is displayed, the frame index,
+# flip time, and time since last movie frame flip will be printed
+reportNDroppedFrames = 10
+
+import os
+import sys
+import threading
+import weakref
+
+# Ensure setting pyglet.options['debug_gl'] to False is done prior to any
+# other calls to pyglet or pyglet submodules, otherwise it may not get picked
+# up by the pyglet GL engine and have no effect.
+# Shaders will work but require OpenGL2.0 drivers AND PyOpenGL3.0+
+import pyglet
+pyglet.options['debug_gl'] = False
+GL = pyglet.gl
+
+
+import psychopy
+from psychopy import core, logging
+
+from psychopy.tools.arraytools import val2array
+from psychopy.tools.attributetools import logAttrib, setAttribute
+from psychopy.tools.filetools import pathToString
+from psychopy.visual.basevisual import BaseVisualStim, ContainerMixin
+from psychopy.clock import Clock
+from psychopy.constants import FINISHED, NOT_STARTED, PAUSED, PLAYING, STOPPED
+from psychopy.tools.monitorunittools import convertToPix
+
+import ctypes
+import numpy
+
+try:
+    import vlc
+except Exception as err:
+    if sys.maxsize == 9223372036854775807:
+        bits = 64
+    else:
+        bits = 32
+    if "wrong architecture" in err:
+        msg = ("Failed to import vlc module for MovieStim2.\n"
+               "You're using %i-bit python. Is your VLC install the same?"
+               % bits)
+        raise OSError(msg)
+    else:
+        raise err
+
+
+
+class TexturedRect:
```

Comment:
I think if your class is a subclass of PsychoPy base classes (probably BaseVisualStim, ContainerMixin, TextureMixin) you won't need your own Texture-rendering code and you'll get various additional stuff for free like the stimulus knowing if it overlaps other stimuli (or the mouse)

Reply:
That would solve the question of flipHoriz/flipVert too because they come for free with TextureMixin